test/utils.ts

```diff
@@ -0,0 +1,66 @@
+
+import { Stats, stat, readFile, writeFile } from 'fs';
+import * as glob from 'glob';
+import { ncp } from 'ncp';
+import * as path from 'path';
+import * as pify from 'pify';
+import { ChildProcess, ForkOptions, fork, SpawnOptions, spawn } from 'child_process';
+import * as once from 'once';
+import * as tmp from 'tmp';
+import * as rimraf from 'rimraf';
+
+export const BUILD_DIRECTORY = 'build';
+
+export const globP: (pattern: string) => Promise<string[]> = pify(glob);
+export const ncpP: (src: string, dest: string) => Promise<void> = pify(ncp);
+export const readFileP: (path: string, encoding?: string) => Promise<Buffer|string> = pify(readFile);
+export const writeFileP: (path: string, data: string, encoding?: string) => Promise<void> = pify(writeFile);
+export const statP: (path: string) => Promise<Stats> = pify(stat);
+export const tmpDirP: () => Promise<string> = pify(tmp.dir);
+export const rimrafP: (f: string) => Promise<void> = pify(rimraf);
+
+export function nodule(nodule: string) {
+  return path.relative(BUILD_DIRECTORY, `node_modules/${nodule}`);
+}
+
+export function existsP(path: string): Promise<boolean> {
+  return statP(path).then(
+    () => Promise.resolve(true),
+    () => Promise.resolve(false)
+  );
+}
+
+function promisifyChildProcess(childProcess: ChildProcess): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const exit = (err?: Error) => once(() => err ? reject(err) : resolve())();
+    childProcess.on('error', exit);
+    childProcess.on('close', (code) => {
+      if (code === 0) {
+        exit();
+      } else {
+        exit(new Error(`Process ${childProcess.pid} exited with code ${code}.`));
+      }
+    });
+  });
+}
+
+export function spawnP(command: string, args?: string[], options?: SpawnOptions, log?: (text: string) => void): Promise<void> {
```

Comment:
This just reminds me something --  can we also config the gts to fix test as well as src. In that way we don't have a line too long.